xbmc/video/VideoInfoScanner.cpp

```diff
@@ -2373,41 +2382,58 @@ namespace KODI::VIDEO
+                                           const std::string& strPath,
+                                           const std::string& parentPath)
   {
     CFileItemList items;
+    CFileItemList parentItems;
+
     // don't try to fetch anything local with plugin source
     if (!URIUtils::IsPlugin(strPath))
     {
-      std::string actorsDir = URIUtils::AddFileToFolder(strPath, ".actors");
+      const std::string actorsDir = URIUtils::AddFileToFolder(strPath, ".actors");
+      const std::string parentActorsDir = URIUtils::AddFileToFolder(parentPath, ".actors");
       if (CDirectory::Exists(actorsDir))
         CDirectory::GetDirectory(actorsDir, items, ".png|.jpg|.tbn", DIR_FLAG_NO_FILE_DIRS |
                                  DIR_FLAG_NO_FILE_INFO);
+      if (!parentPath.empty() && actorsDir != parentActorsDir &&
+          CDirectory::Exists(parentActorsDir))
+        CDirectory::GetDirectory(parentActorsDir, parentItems, ".png|.jpg|.tbn",
+                                 DIR_FLAG_NO_FILE_DIRS | DIR_FLAG_NO_FILE_INFO);
     }
-    for (std::vector<SActorInfo>::iterator i = actors.begin(); i != actors.end(); ++i)
+    for (auto& i : actors)
     {
-      if (i->thumb.empty())
+      if (i.thumb.empty())
       {
-        std::string thumbFile = i->strName;
+        std::string thumbFile = i.strName;
         StringUtils::Replace(thumbFile, ' ', '_');
-        for (int j = 0; j < items.Size(); j++)
-        {
-          std::string compare = URIUtils::GetFileName(items[j]->GetPath());
-          URIUtils::RemoveExtension(compare);
-          if (!items[j]->m_bIsFolder && compare == thumbFile)
-          {
-            i->thumb = items[j]->GetPath();
-            break;
-          }
-        }
-        if (i->thumb.empty() && !i->thumbUrl.GetFirstUrlByType().m_url.empty())
-          i->thumb = CScraperUrl::GetThumbUrl(i->thumbUrl.GetFirstUrlByType());
-        if (!i->thumb.empty())
-          CServiceBroker::GetTextureCache()->BackgroundCacheImage(i->thumb);
+        std::string path{FindActorThumbFile(items, thumbFile)};
+        if (path.empty())
+          path = FindActorThumbFile(parentItems, thumbFile);
+        if (!path.empty())
+          i.thumb = path;
+        else if (path.empty() && !i.thumbUrl.GetFirstUrlByType().m_url.empty())
+          i.thumb = CScraperUrl::GetThumbUrl(i.thumbUrl.GetFirstUrlByType());
+        if (!i.thumb.empty())
+          CServiceBroker::GetTextureCache()->BackgroundCacheImage(i.thumb);
       }
     }
   }
 
+  std::string CVideoInfoScanner::FindActorThumbFile(const CFileItemList& items,
+                                                    const std::string& thumbFile)
+  {
+    for (const auto& file : items.GetList())
+    {
+      std::string compare = URIUtils::GetFileName(file->GetPath());
+      URIUtils::RemoveExtension(compare);
+      if (!file->m_bIsFolder && compare == thumbFile)
+        return file->GetPath();
```

Comment:
How about checking early if the file is actually a folder? Saves doing some unnecessary work

```suggestion
      if (file->m_bIsFolder)
        continue;

      std::string compare = URIUtils::GetFileName(file->GetPath());
      URIUtils::RemoveExtension(compare);
      if (compare == thumbFile)
        return file->GetPath();
```

Reply:
Sure. Done.